src/pybind/mgr/cephadm/tests/test_services.py

```diff
@@ -424,14 +535,21 @@ def test_grafana_initial_admin_pw(self, cephadm_module: CephadmOrchestrator):
                                     '  cookie_samesite = none\n'
                                     '  allow_embedding = true',
                                 'provisioning/datasources/ceph-dashboard.yml':
-                                    '# This file is generated by cephadm.\n'
-                                    'deleteDatasources:\n'
-                                    '\n'
-                                    'datasources:\n'
-                            }
-                    },
-                    [],
-                )
+                                    "# This file is generated by cephadm.\n"
+                                    'deleteDatasources:\n\n'
+                                    "  - name: 'Loki'\n"
+                                    '    orgId: 2\n\n'
+                                    'datasources:\n\n'
+                                    "  - name: 'Loki'\n"
+                                    "    type: 'loki'\n"
+                                    "    access: 'proxy'\n"
+                                    '    orgId: 2\n'
+                                    "    url: 'http://[1::4]:3100'\n"
+                                    '    basicAuth: false\n'
+                                    '    isDefault: true\n'
+                                    '    editable: false',
```

Comment:
What about using heredoc syntax? (and it's better not to mix quotes with double quotes):
```suggestion
                                    dedent("""\
                                    # This file is generated by cephadm.
                                    deleteDatasources:
                                    
                                      - name: 'Loki'
                                    ...
                                        editable: false
                                        """),
```

Reply:
Sure